corehq/form_processor/backends/couch/update_strategy.py

```diff
@@ -315,20 +315,18 @@ def _apply_attachments_action(self, attachment_action, xform=None):
         """
         # the actions and _attachment must be added before the first saves can happen
         # todo attach cached attachment info
-        def fetch_attachment(name):
-            if xform and 'data' in xform._attachments[name]:
-                assert xform.form_id == attachment_action.xform_id
-                return base64.b64decode(xform._attachments[name]['data'])
-            else:
-                return XFormInstance.get_db().fetch_attachment(attachment_action.xform_id, name)
+        def fetch_attachment(name, _form=[]):
```

Comment:
yeah, it's a tradeoff between adding another variable outside of this function or doing this hacky thing with a (private, indicated by leading underscore) parameter with default value. The amount of code you need to read to understand how it's being used is only 4 lines, so seemed like a reasonable tradeoff to me. I'll change it if you think it's too clever.

Good point on using `xform`, I'll fix that part.